geomesa-spark/geomesa_pyspark/src/main/python/geomesa_pyspark/__init__.py

```diff
@@ -0,0 +1,104 @@
+    os.environ['PYSPARK_PYTHON'] = sys.executable
+    
+    spark_classpath = os.environ.get('SPARK_CLASSPATH', '')
+    if len(spark_classpath) > 1:
+        spark_jars = spark_classpath.split(',')
+        spark_jars_set = {jar for jar in spark_jars}
+        spark_jars = spark_jars + [jar for jar in jars if jar not in spark_jars_set]
+    else:
+        spark_jars = jars
+    os.environ['SPARK_CLASSPATH'] = ','.join(spark_jars)
+    
+    spark_home = process_spark_home(spark_home)
+    
+    pyspark_dir = os.path.join(spark_home, 'python')
+    pyspark_lib_dir = os.path.join(pyspark_dir, 'lib')
+    pyspark_lib_zips = glob.glob(os.path.join(pyspark_lib_dir, '*.zip'))
+    sys_path_set = {path for path in sys.path}
+    for pyspark_lib_zip in pyspark_lib_zips:
+        if pyspark_lib_zip not in sys_path_set and os.path.basename(pyspark_lib_zip) != 'pyspark.zip':
+            sys.path.insert(1, pyspark_lib_zip)
+    if pyspark_dir not in sys_path_set:
+        sys.path.insert(1, pyspark_dir)
+    
+    py_files = pyspark_lib_zips + process_executor_packages(packages, tmp_path)
+
+    assert spark_master is 'yarn', 'only yarn master is supported with this release'
+
+    import pyspark
+
+    return (
+        pyspark.SparkConf()
+        .setMaster(spark_master)
+        .set('spark.yarn.dist.jars', ','.join(jars))
+        .set('spark.yarn.dist.files', ','.join(py_files + files))
+        .setExecutorEnv('PYTHONPATH', ":".join(map(os.path.basename, py_files)))
+        .setExecutorEnv('PYSPARK_PYTHON', sys.executable)
+    )
+
+        
+def process_spark_home(spark_home):
+    if spark_home is None:
+        spark_home = os.environ.get('SPARK_HOME', None)
+    assert spark_home is not None, 'unable to resolve SPARK_HOME'
+    assert os.path.isdir(spark_home), '%s is not a directory' % spark_home
+    os.environ['SPARK_HOME'] = spark_home
+    return spark_home
+
+
+def process_executor_packages(executor_packages, tmp_path=None):
+    if tmp_path is None:
+        version_info = sys.version_info
+        tmp_path = os.path.join(tempfile.gettempdir(), 'spark-python-%s.%s' % (version_info.major, version_info.minor))
+    if not os.path.isdir(tmp_path):
+        os.makedirs(tmp_path)
+    driver_packages = {module for _, module, package in pkgutil.iter_modules() if package is True}
+    executor_files = []
+    for executor_package in executor_packages:
+        
+        if executor_package not in driver_packages:
+            raise ImportError('unable to locate ' + executor_package + ' installed in driver')
+
+        package = sys.modules.get(executor_package, None)
+        if package is None:
+            package = pkgutil.get_loader(executor_package).load_module(executor_package)
+
+        package_path = os.path.dirname(package.__file__)
+        package_root = os.path.dirname(package_path)
+
+        if package_root[-4:].lower() in PACKAGE_EXTENSIONS:
```

Comment:
Can you add a comment for the '-4'? 

I *think* this is doing something like cutting off the last few things in the classpath...  Something about this seems fragile...